e2e/analysis/analysis_build_locally_test.go

```diff
@@ -0,0 +1,127 @@
+	. "github.com/onsi/gomega"
+
+	"github.com/ZupIT/horusec-devkit/pkg/utils/logger"
+	"github.com/ZupIT/horusec/config"
+	"github.com/ZupIT/horusec/e2e/analysis"
+	customimages "github.com/ZupIT/horusec/internal/entities/custom_images"
+	"github.com/ZupIT/horusec/internal/utils/testutil"
+)
+
+var _ = Describe("Run a complete horusec analysis when build tools locally", func() {
+	tmpDir := CreateHorusecConfigAndReturnTMPDirectory()
+
+	Describe("Running e2e tests when build tools locally", func() {
+		allTestsCases := analysis.NewTestCase(tmpDir)
+		isWindows := runtime.GOOS == "windows"
+
+		for idx := range allTestsCases {
+			testCase := allTestsCases[idx]
+			if isWindows && testCase.RequiredDocker {
+				logger.LogInfo("Tool ignored because is required docker: ", testCase.Tool, fmt.Sprintf("[%v/%v]", idx+1, len(allTestsCases)))
+				continue
+			}
+			logger.LogInfo("Preparing test for run e2e build locally: ", testCase.Tool, fmt.Sprintf("[%v/%v]", idx+1, len(allTestsCases)))
+			session, err := testCase.RunAnalysisTestCase()
+			if err != nil {
+				Fail(fmt.Sprintf("Error on run analysis Test Case when build: %v", err))
+			}
+			session.Wait(testutil.AverageTimeoutAnalyzeForExamplesFolder)
+			testCase.Command.Output = string(session.Out.Contents())
+			testCase.Command.ExitCode = session.ExitCode()
+
+			It(fmt.Sprintf("In Build tools locally execute command without error for: %s", testCase.Tool), func() {
+				Expect(testCase.Command.ExitCode).Should(Equal(0))
+			})
+
+			It(fmt.Sprintf("In Build tools locally validate is outputs expected exists on tool: %s", testCase.Tool), func() {
+				for _, outputExpected := range testCase.Expected.OutputsContains {
+					Expect(testCase.Command.Output).Should(
+						ContainSubstring(outputExpected),
+						fmt.Sprintf("The output [%s] not exist in output", outputExpected),
+						testCase.Command.Output,
+					)
+				}
+			})
+
+			It(fmt.Sprintf("In Build tools locally validate is outputs not expected exists on tool: %s", testCase.Tool), func() {
+				for _, outputNotExpected := range testCase.Expected.OutputsNotContains {
+					Expect(testCase.Command.Output).ShouldNot(
+						ContainSubstring(outputNotExpected),
+						fmt.Sprintf("The output [%s] exist in output", outputNotExpected),
+						testCase.Command.Output,
+					)
+				}
+			})
+
+			if CurrentGinkgoTestDescription().Failed {
+				logger.LogInfo("Tool finished your analysis with status of failed: ", testCase.Tool, fmt.Sprintf("[%v/%v]", idx+1, len(allTestsCases)))
+				os.Exit(1)
+			}
+			logger.LogInfo("Tool finished your analysis with status of success: ", testCase.Tool, fmt.Sprintf("[%v/%v]", idx+1, len(allTestsCases)))
+		}
+	})
+
+	AfterEach(func() {
+		if err := os.RemoveAll(tmpDir); err != nil {
+			Fail(fmt.Sprintf("Error on remove tmp folder: %v", err))
+		}
+	})
+})
+
+func CreateHorusecConfigAndReturnTMPDirectory() string {
+	tmpDir := filepath.Join(os.TempDir(), "horusec-analysis-e2e-"+uuid.NewString())
```

Comment:
I think that here would be better use the `TempDir()` from [`GinkgoT()`](https://pkg.go.dev/github.com/onsi/ginkgo#GinkgoT). Using this approach we don't need to deal with clean up temp directory.

`filepath.Join(GinkgoT().TempDir(), "horusec-analysis-e2e-"+uuid.NewString())`

Reply:
sounds good for me, fixed

Reply:
in my tests i found one problem, `GinkgoT().TempDir()` is currectly directory and the library not remove after test finish, then the user can up your code with files of the tests, i beleave we can keep os.TempDir for now

Reply:
This is weird. `GinkoT()` should return a `*testing.T` and according to [`TempDir`](https://pkg.go.dev/testing#T.TempDir) docs the temp directory is automatically removed by Cleanup

Reply:
in Ginkgo is not possible :(